ast/ast.go

```diff
@@ -64,6 +64,10 @@ type ExprNode interface {
 	SetValue(val interface{})
 	// GetValue gets value of the expression.
 	GetValue() interface{}
+	// SetDatum set datum to the expression.
+	SetDatum(datum *types.Datum)
```

Comment:
In that way, does the Datum will be copy twice? once in the pass of parm, once in the copy it to expr.